rust/agama-lib/share/profile.schema.json

```diff
@@ -671,19 +671,45 @@
         }
       ]
     },
-    "search": {
-      "title": "Search options",
+    "searchName": {
+      "title": "Device name",
+      "type": "string",
+      "examples": ["/dev/vda", "/dev/disk/by-id/ata-WDC_WD3200AAKS-75L9"]
+    },
+    "searchByName": {
+      "title": "Search by name condition",
       "type": "object",
       "additionalProperties": false,
       "required": ["name"],
       "properties": {
         "name": {
-          "title": "Device name",
-          "type": "string",
-          "examples": ["/dev/vda"]
+          "$ref": "#/$defs/searchName"
         }
       }
     },
+    "search": {
```

Comment:
The document [storage_auto](https://github.com/openSUSE/agama/blob/master/doc/auto_storage.md) suggests two possible syntaxes for the search condition:

Option A:

~~~json 
{
  "search": {
    "condition": { "property": "name", "value": "/dev/sda2" }
  }
}
~~~

Option B:

~~~json
{
  "search": {
    "condition": { "name": "/dev/sda2" }
  }
}
~~~

The option B was chosen in this PR because the following reasons:

Reply:
There is third option to be discussed:

~~~json
{
  "search": {
  }
}
~~~